Remove all deprecated code

rafaelfranca opened the pull request:
Remove all code deprecated in 7.1.

p8 commented:
I removed `secrets` in the following PR: https://github.com/rails/rails/pull/49618.
There were more changes than expected.

rafaelfranca commented:
Thanks! I'll apply your commit to my branch

zzak commented:
Should we remove `active_support.cache_format_version = 6.1` as well?

```
DEPRECATION WARNING: Support for `config.active_support.cache_format_version = 6.1` has been deprecated and will be removed in Rails 7.2.
```

rafaelfranca commented:
Yes. I'll remove every single deprecation.

nvasilevski commented on activerecord/lib/active_record/attribute_methods.rb:
```diff
@@ -88,31 +88,15 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
-            ActiveRecord.deprecator.warn(
-              "#{self} model aliases `#{old_name}`, but `#{old_name}` is not an attribute. " \
-              "Starting in Rails 7.2, alias_attribute with non-attribute targets will raise. " \
+          # We only need to raise once, so we issue it when defining the original reader method.
+          is_reader_method = target_name == old_name
+
+          if is_reader_method
```
We don't need this condition anymore. We needed it to avoid raising more than one deprecation warning but with `raise` it  shouldn't matter since `raise` will halt further method definitions

nvasilevski commented on activerecord/lib/active_record/attribute_methods.rb:
```diff
@@ -88,31 +88,15 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
-            "You may want to additionally define `#{method_name}` to preserve the current behavior."
-          )
+
           super
```
I believe our intention was to always call `define_proxy_call` for all kinds of `alias_attribute` usages. So basically the code should either:
1. Raise if the application tries to alias a non-attribute thing
2. Proceed to `define_proxy_call`

rafaelfranca replied:
Good point. Now I got sniped to understand why `super` and `define_proxy_call` and so similar :joy:. I'll fix this commit.

rafaelfranca committed:
Remove deprecated `config.active_storage.replace_on_assign_to_many`

rafaelfranca committed:
Remove deprecated `config.active_storage.silence_invalid_content_types_warning`

rafaelfranca committed:
Remove deprecated support for the pre-Ruby 2.4 behavior of `to_time`

When `to_time_preserves_timezone` was set to `false`, `to_time`
would return a `Time` object with local timezone. This behavior
has been removed.

rafaelfranca committed:
Remove deprecated `:pool_size` and `:pool_timeout` options for the cache storage

rafaelfranca committed:
Remove deprecated support for `config.active_support.cache_format_version = 6.1`

rafaelfranca committed:
Remove deprecated constants `ActiveSupport::LogSubscriber::CLEAR` and `ActiveSupport::LogSubscriber::BOLD`

rafaelfranca committed:
Remove deprecated support to bolding log text with positional boolean in `ActiveSupport::LogSubscriber#color`

rafaelfranca committed:
Remove deprecated `config.active_support.disable_to_s_conversion`

rafaelfranca committed:
Remove deprecated `config.active_support.remove_deprecated_time_with_zone_name`

rafaelfranca committed:
Remove deprecated `config.active_support.use_rfc4122_namespaced_uuids`

rafaelfranca committed:
Remove deprecated support to passing `Dalli::Client` instances to `MemCacheStore`

rafaelfranca committed:
Remove deprecated `#to_default_s` from `Array`, `Date`, `DateTime` and `Time`

rafaelfranca committed:
Update deprecator horizon for 7.3

This change was never released on 7.1, so we can't remove it on 7.2.

rafaelfranca committed:
Remove deprecated `SafeBuffer#clone_empty`

rafaelfranca committed:
Remove deprecated `ActiveSupport::Deprecation` delegation to instance

rafaelfranca committed:
Remove deprecated support to call the following methods without passing a deprecator

  - `deprecate`
  - `deprecate_constant`
  - `ActiveSupport::Deprecation::DeprecatedObjectProxy.new`
  - `ActiveSupport::Deprecation::DeprecatedInstanceVariableProxy.new`
  - `ActiveSupport::Deprecation::DeprecatedConstantProxy.new`
  - `assert_deprecated`
  - `assert_not_deprecated`
  - `collect_deprecations`

rafaelfranca committed:
Remove deprecated `ActiveSupport::Notifications::Event#children` and  `ActiveSupport::Notifications::Event#parent_of?`

rafaelfranca committed:
Remove unnecessary deprecation silencing in cache_store_compression_behavior.rb

rafaelfranca committed:
Upgrade prism

rafaelfranca committed:
Remove deprecation on calling `assert_equal` with `nil`

rafaelfranca committed:
Remove deprecated `Rails.application.secrets`

rafaelfranca committed:
Remove deprecated `Rails.config.enable_dependency_loading`

rafaelfranca committed:
Remove deprecated `find_cmd_and_exec` console helper

rafaelfranca committed:
Remove deprecated `Rails::Generators::Testing::Behaviour`
